src/glue.c

```diff
@@ -561,6 +563,33 @@ void genhelpers(Module *m, bool iscomdat)
 }
 
 /**************************************
+ * Bugzilla 14828: If some template instantiations require
+ * non-root module helpers, they should be placed in
+ * separated object file.
+ */
+void genHelpersObjFile(Library *library, Module *m)
+{
+    assert(!m->isRoot());
+
+    lastmname = m->srcfile->toChars();
+
+    obj_start(lastmname);
+
+    OutBuffer idbuf;
+    idbuf.printf("%s", m->ident->toChars());
+    char *idstr = idbuf.peekString();
+
+    idbuf.data = NULL;
+    Identifier *id = Identifier::create(idstr, TOKidentifier);
```

Comment:
> id seems unused here

Ouch, the code was mostly copy-pasted from `obj_write_deferred`. I'll remove it.

> Should prolly follow the naming scheme used here.

If a module file name is `a.d`, its ModuleInfo and module helpers will be stored in `a.obj` with `-lib` or `-multiobj` switch. In here, for a non-root module `b.d`, the module helpers will be stored in `b.obj`. The naming scheme (without number suffix) is intentional.